packages/frontend-api/src/Controller/SocialNetworkController.php

```diff
@@ -126,7 +114,7 @@ protected function getRefererUrl(
         $refererUrl = $refererUrl ?? $homepageUrl;
         $request->getSession()->remove(self::REFERER_URL);
         $refererUrl = str_replace($this->domain->getUrl(), '', $refererUrl);
-        $url = '/social-login?redirect=' . $refererUrl;
+        $url = sprintf('%s?redirect=%s', $this->generateUrl('front_social_network_login_page', referenceType: UrlGeneratorInterface::RELATIVE_PATH), $refererUrl);
```

Comment:
Why don't we use generateUrl parameters for the URL instead of the sprintf? And it will be escaped correctly.

Reply:
that is a good idea :pray: 14c125f5c1